### ISSUE TITLE
推理相关问题

### ISSUE THREAD

**Issue by Paulisca8** [RickyL-2000/ROSVOT#10]
您好。在推理midi的时候，会自动将潜在的word_boundary拆开，这个模式是否能够禁用呢？

因为我不希望再改掉我的durations序列了，这样我又得倒回去重新对音素序列进行调整。我希望输出的midi序列能够与我输入的durations等长。

**Comment by RickyL-2000** [comment]
我理解你的意思是不希望存在一字多音的情况，希望是字的数量和音的数量相等。我认为有两种解决方法。

一个简单而直接的方法是，将[这一行](https://github.com/RickyL-2000/ROSVOT/blob/3c8332bf43adae35f6e4d64971862f2f6139b310/modules/rosvot/rosvot.py#L236C9-L236C88)从原来的：

```python
note_lengths, note_logits, note_pred = self.pitch_decoder(feat, note_bd, train)
```

改成：

```python
note_lengths, note_logits, note_pred = self.pitch_decoder(feat, word_bd, train)
```

原理是，直接override，将 word boundary 视为 note boundary，那么理论上这样产出的MIDI和word的数量是相等的。

第二个方法更简单，修改 --thr 这个 flag 的值为1，这个 threshold 用于决定某个 logits 是否为 note boundary。当 threshold==1 的时候所有 note boundary 都将不存在，那么就完全根据 word boundary 来决定。

但是我强烈不建议这么做，因为一字多音是歌唱中非常常见的技巧。如果我理解错了，请纠正我。

**Comment by Paulisca8** [comment]
感谢您的回复，问题已经解决。

我所提供的 duration_array 是我人工将同一音素不同音高进行分割后得到的结果，也就是说我已经人工对一字多音的情况分割了note boundary，不再需要本项目中的自动化分割。我也会尝试对比自动化分割与人工分割的效果，再次感谢。
